programs/ui-wrapper/src/processors/shared.rs

```diff
@@ -0,0 +1,345 @@
+            // one of the orders fully matches as it is being placed. We only
+            // know that the indices are right, not the actual orders there.
+            node.get_mut_value()
+                .set_price(core_resting_order.get_price());
+            node.get_mut_value()
+                .set_is_bid(core_resting_order.get_is_bid());
+        }
+        let mut orders_tree: RedBlackTree<WrapperOpenOrder> =
+            RedBlackTree::<WrapperOpenOrder>::new(wrapper_dynamic_data, orders_root_index, NIL);
+        for to_remove_index in to_remove_indices.iter() {
+            orders_tree.remove_by_index(*to_remove_index);
+        }
+        orders_root_index = orders_tree.get_root_index();
+
+        let wrapper_fixed: &mut ManifestWrapperStateFixed = get_mut_helper(fixed_data, 0);
+        let mut free_list: FreeList<UnusedWrapperFreeListPadding> =
+            FreeList::new(wrapper_dynamic_data, wrapper_fixed.free_list_head_index);
+        for open_order_index in to_remove_indices.iter() {
+            // Free the node in wrapper.
+            free_list.add(*open_order_index);
+        }
+        wrapper_fixed.free_list_head_index = free_list.get_head();
+    }
+    let market_info: &mut MarketInfo =
+        get_mut_helper::<RBNode<MarketInfo>>(wrapper_dynamic_data, market_info_index)
+            .get_mut_value();
+    market_info.orders_root_index = orders_root_index;
+
+    // Sync balances
+    let market_info: &mut MarketInfo =
+        get_mut_helper::<RBNode<MarketInfo>>(wrapper_dynamic_data, market_info_index)
+            .get_mut_value();
+    let claimed_seat: &ClaimedSeat =
+        get_helper::<RBNode<ClaimedSeat>>(market_ref.dynamic, market_info.trader_index).get_value();
+    market_info.base_balance = claimed_seat.base_withdrawable_balance;
+    market_info.quote_balance = claimed_seat.quote_withdrawable_balance;
+    let quote_volume_difference = claimed_seat
+        .quote_volume
+        .wrapping_sub(market_info.quote_volume);
+    market_info.quote_volume_unpaid = market_info
+        .quote_volume_unpaid
+        .saturating_add(quote_volume_difference);
+    market_info.quote_volume = claimed_seat.quote_volume;
+    market_info.last_updated_slot = Clock::get().unwrap().slot as u32;
+
+    Ok(())
+}
+
+pub(crate) fn get_market_info_index_for_market(
+    wrapper_state: &WrapperStateAccountInfo,
+    market: &Pubkey,
+) -> DataIndex {
+    let mut wrapper_data: RefMut<&mut [u8]> = wrapper_state.info.try_borrow_mut_data().unwrap();
+    let (fixed_data, wrapper_dynamic_data) =
+        wrapper_data.split_at_mut(size_of::<ManifestWrapperStateFixed>());
+
+    let wrapper_fixed: &ManifestWrapperStateFixed = get_helper(fixed_data, 0);
+    let market_infos_tree: MarketInfosTree = MarketInfosTree::new(
+        wrapper_dynamic_data,
+        wrapper_fixed.market_infos_root_index,
+        NIL,
+    );
+
+    // Just need to lookup by market key so the rest doesnt matter.
+    let market_info_index: DataIndex =
```

Comment:
this should not have changed

Reply:
what do you mean?

Reply:
thought it was a param in a version I was looking at